src/main/app/controller/courts/AddressController.ts

```diff
@@ -0,0 +1,276 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {AxiosError} from 'axios';
+import {CSRF} from '../../../modules/csrf';
+import {
+  AddressType,
+  AddressValidationResult,
+  CourtAddress,
+  DisplayAddress,
+  DisplayCourtAddresses
+} from '../../../types/CourtAddress';
+import {CourtAddressPageData} from '../../../types/CourtAddressPageData';
+import {SelectItem} from '../../../types/CourtPageData';
+
+@autobind
+export class AddressController {
+
+  getAddressTypesError = 'A problem occurred when retrieving the court address types.';
+  getAddressesError = 'A problem occurred when retrieving the court addresses.';
+  updateAddressError = 'A problem occurred when saving the court addresses.';
+  multipleVisitAddressError = 'Only one visit address is permitted.';
+  typeRequiredError = 'Address Type is required.';
+  addressRequiredError = 'Address is required.';
+  townRequiredError = 'Town is required.';
+  invalidPostcodeError = 'Postcode is invalid.';
+  postcodeMissingError = 'Postcode is required.';
+  postcodeNotFoundError = 'Postcode entered could not be found.';
+  writeToUsAddressType = 'Write to us';
+  visitOrContactUsAddressType = 'Visit or contact us';
+  primaryAddressPrefix = 'Primary Address: ';
+  secondaryAddressPrefix = 'Secondary Address: ';
+
+  public async get(req: AuthedRequest, res: Response): Promise<void> {
+    await this.render(req, res);
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+    const addresses: DisplayCourtAddresses = {
+      primary: req.body.primary,
+      secondary: req.body.secondary,
+    };
+    const writeToUsTypeId = req.body.writeToUsTypeId;
+
+    // Validate token
+    if(!CSRF.verify(req.body._csrf)) {
+      await this.render(req, res, false, addresses, [this.updateAddressError]);
+      return;
+    }
+
+    // Validate addresses
+    const primaryValidationResult = this.validateAddress(addresses.primary, true);
```

Comment:
Food for thought, could lines 52 to either 56, or even 63 be in a separate method? Up to you